packages/core/src/error/index.ts

```diff
@@ -96,14 +88,4 @@ export class MastraError extends Error {
   }
 }
 
-const error = new MastraError({
-  id: 'BASE_TEST_001',
-  text: 'This is a base test error',
-  domain: Domain.AGENT,
-  category: ErrorCategory.UNKNOWN,
-  details: {
-    tset: 'lalal',
-  },
-});
-
-console.log(error.toJSON());
+export class MastraError extends MastraBaseError<ErrorDomain, ErrorCategory> {}
```

Comment:
always stringify enums so you can use the values instead of the real enum
```suggestion
export class MastraError extends MastraBaseError<`${ErrorDomain}`, `${ErrorCategory}`> {}
```